test/lint/lint-shell.sh

```diff
@@ -21,11 +21,40 @@ if ! command -v shellcheck > /dev/null; then
     exit 0
 fi
 
+EXIT_CODE=0
+
 # Disabled warnings:
 disabled=(
     SC2046 # Quote this to prevent word splitting.
     SC2086 # Double quote to prevent globbing and word splitting.
     SC2162 # read without -r will mangle backslashes.
 )
-shellcheck -e "$(IFS=","; echo "${disabled[*]}")" \
-    $(git ls-files -- "*.sh" | grep -vE 'src/(secp256k1|univalue)/')
+EXCLUDE="--exclude=$(IFS=','; echo "${disabled[*]}")"
+
+disabled_gitian=(
+    SC2094 # Make sure not to read and write the same file in the same pipeline.
+    SC2129 # Consider using { cmd1; cmd2; } >> file instead of individual redirects.
+    SC2230 # which is non-standard. Use builtin 'command -v' instead.
+)
+EXCLUDE_GITIAN=${EXCLUDE}",$(IFS=','; echo "${disabled_gitian[*]}")"
+
+if ! shellcheck "$EXCLUDE" $(git ls-files -- '*.sh' | grep -vE 'src/(leveldb|secp256k1|univalue)/'); then
+    EXIT_CODE=1
+fi
+
+if ! command -v yq > /dev/null; then
+    echo "Skipping Gitian desriptor scripts checking since yq is not installed."
+    exit $EXIT_CODE
+fi
+
+for descriptor in $(git ls-files -- 'contrib/gitian-descriptors/*.yml')
+do
+    echo
+    echo "$descriptor"
+    SCRIPT=$'#!/bin/bash\n'$(yq -r .script "$descriptor")
```

Comment:
```suggestion
    SCRIPT=$'#!/usr/bin/env bash\n'$(yq -r .script "$descriptor")
```

Reply:
It just mimics the way [gitian-builder/bin/gbuild](https://github.com/devrandom/gitian-builder/blob/cd8aa53c4e8870fa6be614a653024bae094c2f22/bin/gbuild#L128) makes shell script:
```
  File.open("var/build-script", "w") do |script|
    script.puts "#!/bin/bash"
```

Reply:
Ah! Makes sense to do it this way, could you add a comment?

Reply:
Sure!

Reply:
Done.